lib/src/scripts/books.dart

```diff
@@ -85,16 +88,42 @@ class BookInfo {
     required this.maturityRating,
     required this.pageCount,
     required this.publishedDate,
+    required this.rawPublishedDate,
     required this.ratingsCount,
   });
 
   static BookInfo fromJson(Map<String, dynamic> json) {
     final publishedDateArray =
         ((json['publishedDate'] as String?) ?? '0000-00-00').split('-');
-    final year = int.parse(publishedDateArray[0]);
-    final month = int.parse(publishedDateArray[1]);
-    final day = int.parse(publishedDateArray[2]);
-    final publishedDate = DateTime(year, month, day);
+
+    // initialize date
+    int year = 0;
+    int month = 1;
+    int day = 1;
+
+    // now test the date string
+    if(publishedDateArray.length==1) {
+      // assume we have only the year
+      year = int.parse(publishedDateArray[0]);
+    }
+    if(publishedDateArray.length==2) {
+      // assume we have the year and maybe the month (this could be just a speculative case)
+      year = int.parse(publishedDateArray[0]);
+      month = int.parse(publishedDateArray[1]);
+    }
+    if(publishedDateArray.length==3) {
+      // assume we have year-month-day
+      year = int.parse(publishedDateArray[0]);
+      month = int.parse(publishedDateArray[1]);
+      day = int.parse(publishedDateArray[2]);
+    }
+
+    // initialize datetime variable
+    DateTime publishedDate = DateTime(0);
```

Comment:
Wouldn't it be better to leave it `null` if no date was found, instead of 0?

Reply:
But then it have to be DateTime? publishedDate for nullSafe

Reply:
I have resolved it and now publishedDate could be null